mergify_engine/tests/unit/test_command.py

```diff
@@ -84,3 +92,602 @@ def test_command_loader_wuth_defaults():
+        return MockResponseWithoutPermissions()
+
+    monkeypatch.setattr(client, "post", mock_post)
+
+    await handle(
+        ctxt=ctxt,
+        mergify_config={},
+        comment="@mergifyio something",
+        user=None,
+        rerun=True,
+    )
+
+    assert len(http_calls) == 1
+    assert (
+        "Sorry but I didn't understand the command." in http_calls[0][1]["json"]["body"]
+    )
+
+
+@pytest.mark.asyncio
+async def test_run_command_with_wrong_user(redis_cache, monkeypatch):
+
+    client = github.aget_client(owner_name="Mergifyio", owner_id=123)
+
+    sub = subscription.Subscription(
+        redis_cache,
+        123,
+        True,
+        "",
+        {},
+        frozenset({}),
+    )
+
+    installation = context.Installation(123, "Mergifyio", sub, client, redis_cache)
+
+    repository = context.Repository(installation, "demo", 123)
+
+    ctxt = await context.Context.create(
+        repository,
+        {
+            "number": 789,
+            "state": "open",
+            "title": "Amazing new feature",
+            "user": {
+                "login": "octocat",
+                "id": 1,
+            },
+            "mergeable_state": "ok",
+            "merged_by": None,
+            "merged": None,
+            "merged_at": None,
+            "base": {
+                "sha": "sha",
+                "ref": "main",
+                "user": {
+                    "login": {
+                        "Mergifyio",
+                    },
+                },
+                "repo": {
+                    "name": "demo",
+                    "private": False,
+                    "owner": {
+                        "login": "Mergifyio",
+                        "id": 123,
+                    },
+                    "permissions": {
+                        "admin": False,
+                        "push": False,
+                        "pull": True,
+                    },
+                },
+            },
+            "requested_teams": [
+                {
+                    "id": 1,
+                    "name": "Justice League",
+                    "slug": "justice-league",
+                    "description": "A great team.",
+                    "privacy": "closed",
+                    "permission": "admin",
+                }
+            ],
+        },
+        [],
+    )
+
+    user = github_types.GitHubAccount(
+        {
+            "id": 666,
+            "login": "wall-e",
+            "type": "Bot",
+            "avatar_url": "https://avatars.githubusercontent.com/u/583231?v=4",
+        },
+    )
+
+    http_calls = []
+
+    async def mock_post(*args, **kwargs):
+        http_calls.append((args, kwargs))
+        return MockResponseWithoutPermissions()
+
+    monkeypatch.setattr(client, "post", mock_post)
+
+    permission_calls = []
+
+    async def mock_permission(*args, **kwargs):
+        permission_calls.append((args, kwargs))
+        return MockResponseWithoutPermissions()
```

Comment:
If you change has_write_permission(), you should return the same thing, here a bool.

More generally to make this test more robust you should only monkeypatch/mock external stuff, here the HTTP client.
This will allow us to also test the logic inside has_write_permission.

unittest should only replaces thing that need to access to external resources (here GitHub API calls).